src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -555,24 +555,20 @@ class G1ScanHRForRegionClosure : public HeapRegionClosure {
       return (*card & ToScanMask) == 0;
     }
 
-    static bool is_word_aligned(const void* const addr) {
-      return ((uintptr_t)addr) % sizeof(Word) == 0;
-    }
-
     CardValue* find_first_dirty_card(CardValue* i_card) const {
-      while (!is_word_aligned(i_card)) {
+      while (!CardTable::is_word_aligned(i_card)) {
```

Comment:
pre-existing: This loop assumes initial i_card < _end_card and _end_card is word aligned.  Should be
asserted. Similarly in find_first_non_dirty_card.

Reply:
It's asserted in the constructor.